File: app/src/main/java/me/gabeg/sicksends/SsTradDao.java
package me.gabeg.sicksends;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;
import java.util.List;

/**
 * Data access object for trad problems.
 */
@Dao
public interface SsTradDao
	extends SsProblemDao<SsTrad>
{

	/**
	 * @return All trad problems.
	 */
	@Query("SELECT * FROM trad")
	LiveData<List<SsTrad>> getAll();

}

File: app/src/main/java/me/gabeg/sicksends/db/trad/SsTradDao.kt
package me.gabeg.sicksends.db.trad

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import me.gabeg.sicksends.db.generic.SsGenericProblem
import me.gabeg.sicksends.db.generic.SsGenericProblemDao

/**
 * Data access object for trad problems.
 */
@Dao
interface SsTradDao : SsGenericProblemDao<SsTradProblem>
{

	/**
	 * Delete all trad problems.
	 */
	//@Query("DELETE FROM trad")
	//override suspend fun deleteAll()

	/**
	 * @see SsGenericProblemDao.findProblems
	 */
	@Query("SELECT * FROM trad ${SsGenericProblem.WHERE} ${SsGenericProblem.ORDER_BY}")
	override fun findProblems(
		isOutdoor : Boolean?,
		isProject : Boolean?,
		isFlash : Boolean?) : LiveData<List<SsTradProblem>>

	/**
	 * Get all trad problems.
	 *
	 * @return All trad problems.
	 */
	@Query("SELECT * FROM trad ${SsGenericProblem.ORDER_BY}")
	override fun getAllProblems() : LiveData<List<SsTradProblem>>

}
